boards/arm/xenvm/Kconfig.defconfig

```
@@ -0,0 +1,12 @@
+# Copyright (c) 2020 EPAM Systems
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_XENVM
+
+config BUILD_OUTPUT_BIN
+	default y
+
+config BOARD
+	default "xenvm"
+
```

You should probably add:
```suggestion

config GIC_V2
	default y

config CPU_CORTEX_A72=y
	default y
```

if you need, then you could make a board variant with the A53 SoC.

---

I am a bit concerned this will clutter the number of xenvm variant in Zephyr. We would be looking to at least one board variant per CPU type. You can then multiply it by the number of options that a user can select in the VM configuration file (e.g gic version, device passthrough, RAM...).

Exposing these options to the user seems a better approach to me. Another approach that could be considered is providing a script that would be able to generate the board configuration based on the VM configuration.

---

except that such change will suddenly allow users to reconfigure the SoC on a real board.
That is something that should not be possible, as it may lead to all sorts of confusions later.

Also running Kconfig and change the SoC will most likely impact the generated device tree, and thus the build folder should probably be pristine cleaned.

With this PR, it also becomes possible to select two SoCs (or none), so it basically breaks all rules.

I see the need for more flexibility when working with VMs, and it is possible in the board Kconfig to add a prompt and in that way allow user to select a SoC.
However, such approach must be careful evaluated and tested, especially with regards to DTS.
First step would be to only allow users to select the SoC on first CMake invocation, similar to how it is only possible to select the board, but not change that selection without a pristine build.

Example in `boards/arm/zenvm/Kconfig.defconfig`
```
config CPU_CORTEX_A53
       bool "Allow reconfig (A53)"

config CPU_CORTEX_A72
       bool "Allow reconfig (A72)"
       depends on !CPU_CORTEX_A53
       default y
```

---

Well, I don't even sure that in case of VMs there is a need to divide configuration between SoC and board. Every VM is different in almost all aspects: CPU core type, number of cores, memory size, type of interrupt controller, timer frequency, set of peripherals. I can see only two constant things: SBSA UART and need for Linux image header, so Xen can load and run Zephyr.

Actually, even that is not constant because SBSA might be replaced with Xen virtual conso and Xen image loader supports ELF images (however, only at x86 right now).

So, maybe Zephyr does not need "xenvm" soc and "xenvm" board at all. Or need them as mere examples. You have seen that people already asked "why GICv2?", "why 8.320MHz?". Only answer I have is "because this is what needed on my platform".  

I have two ideas there: I can drop all patches related to "xenvm" and leave only patches with SBSA and Linux Image header, so any user that wants to run Zephyr on Xen can create own board with own configuration.

Or I can rename "xenvm" to "xenvm_rcar_gen3_example", because this is exactly what it is: example showing how to run Zephyr as Xen virtual guest on Renesas RCar Gen3 hardware. This will lift most of the questions about proposed configuration. 

Also, I love Julien's idea about scripts that creates board configuration based on input parameters. But I don't think that we have resources for this.

---

> Every VM is different in almost all aspects: CPU core type, number of cores, memory size, type of interrupt controller, timer frequency, set of peripherals. 

Yes, that is one of the advantages of using VMs for testing.
Unfortunately current board design was not having this is mind when started.

>  I can drop all patches related to "xenvm" 

I think it's great to have a starting point, even if it is fixed.
It makes it easier for users to use the board as base line and adjust it for their own need.

This is not so different from vendor A providing Board B, and some creates their own board with same SoC and using Board B as baseline for a new board, Board B_C.

So let's keep your `xenvm` proposal, but without changing the current way configs are handled.

User can then copy `xenvm` into their own repo and adjust the settings the need.
They can easily add that repo as additional board root.
https://docs.zephyrproject.org/latest/guides/modules.html#build-settings

> Also, I love Julien's idea about scripts that creates board configuration based on input parameters. But I don't think that we have resources for this.

Yes, let's look into this for the future.
But waiting for this should not prevent us from taking this PR in.